### PR TITLE
Fix get_mon_map() for octopus and later

### PR DESCRIPTION
**Pull request by ajkavanagh**
The "ceph mon_status" command seems to have disappeared on octopus and
later, and is replaced by "ceph quorum_status".  This changes the
get_mon_map() function to detect the underlying ceph version and do the
right thing.

Cherry-picked from: 1cf0226b
Related-Bug: https://bugs.launchpad.net/charm-ceph-mon/+bug/1951094

### CODE REVIEW

**Review by lmlg — APPROVED** [review]
LGTM.

**Review by AurelienLourot — APPROVED** [review]
LGTM, thanks, does it need the approval of another core reviewer?

**Review by ChrisMacNaughton — APPROVED** [review]

**Pull request merged**
